pkg/generator/generator.go

```diff
@@ -408,27 +356,19 @@ func renderDockerBuildFile(w io.Writer) error {
 }
 
 func renderCodegenFiles(codegenDir, repoPath, apiDirName, version, projectName string) error {
-	buf := &bytes.Buffer{}
 	bTd := tmplData{
 		ProjectName: projectName,
 	}
-	if err := renderFile(buf, "codegen/boilerplate.go.txt", boilerplateTmpl, bTd); err != nil {
-		return err
-	}
-	if err := writeFileAndPrint(filepath.Join(codegenDir, boilerplate), buf.Bytes(), defaultFileMode); err != nil {
+	if err := renderWriteFile(filepath.Join(codegenDir, boilerplate), "codegen/boilerplate.go.txt", boilerplateTmpl, bTd); err != nil {
 		return err
 	}
 
-	buf = &bytes.Buffer{}
 	ugTd := tmplData{
 		RepoPath:   repoPath,
 		APIDirName: apiDirName,
 		Version:    version,
 	}
-	if err := renderFile(buf, "codegen/update-generated.sh", updateGeneratedTmpl, ugTd); err != nil {
-		return err
-	}
-	return writeFileAndPrint(filepath.Join(codegenDir, updateGenerated), buf.Bytes(), defaultExecFileMode)
```

Comment:
We need the file `codegen/update-generated.sh` to be executable with the `defaultExecFileMode`. Currently with this PR `operator-sdk new ...` fails on the code generation step because the codegen script is not executable:
```
$ operator-sdk new app-operator --api-version=app.example.com/v1 --kind=App
...
(38/39) Wrote github.com/gogo/protobuf@v1.0.0
(39/39) Wrote k8s.io/client-go@kubernetes-1.9.3
Run dep ensure done
Run code-generation for custom resources
Error: failed to perform code-generation for CustomResources: ()
```
Also side note but it's weird that I didn't get an error message on failing to run the code generation command. But we can look into that later
https://github.com/operator-framework/operator-sdk/blob/master/commands/operator-sdk/cmd/generate/k8s.go#L55-L61